### ISSUE TITLE
M0 mask-constrained smoothing sub-optimal?

### ISSUE THREAD

**Issue by HenkMutsaerts** [ExploreASL/ExploreASL#111]
The PDF report for the TestDataSet shows an inhomogeneous M0 after mask-constrained smoothing, perhaps this needs improvement. Below the PDF report from Windows (Matlab 2020a?) server.

![Screenshot 2020-09-01 at 18 38 47](https://user-images.githubusercontent.com/27774254/91881807-60abd900-ec82-11ea-9c6f-79b20fe932a9.png)

Also, this doesn't seem very reproducible, compiled on the same computer this gives:
![Screenshot 2020-09-01 at 18 39 49](https://user-images.githubusercontent.com/27774254/91881919-8507b580-ec82-11ea-8ac9-404f3499d399.png)

But perhaps this is simply because it was run on a low quality option.

@MichaelStritt did you run this test just now on low quality, or on high quality?
If low quality, do you mind rerunning the Windows & compilation both on high quality?

**Comment by MichaelStritt** [comment]
Yes, I thought to find bugs it doesn't make a difference, but I can rerun them on high quality if you want.

**Comment by HenkMutsaerts** [comment]
With your quality=1, we get the following for Windows:
![Screenshot 2020-09-02 at 08 03 53](https://user-images.githubusercontent.com/27774254/91937726-d94a7e00-ecf2-11ea-815b-9d3eab727028.png)
and for compiled:
![Screenshot 2020-09-02 at 08 04 15](https://user-images.githubusercontent.com/27774254/91937748-e6676d00-ecf2-11ea-9478-2563b82ce936.png)
. So this definitely looks better, though the CSF signal still bleeds in, this needs to be masked better.

The CBF reproducibility is now better:
median GM CBF 63.95 vs 63.95 (identical!)
PVC2 GM CBF 86.88 vs **87.08**, a 0.2% change, which is good!
